### ISSUE TITLE
Nothing loads from any sankaku sites, other sites work fine

### ISSUE THREAD

**Issue by cybersoldier8** [Bionus/imgbrd-grabber#728]
When searching on any of the sankaku sites, such as chan.sankakucomplex.com, I get no results at all. Other sites like gelbooru work just fine. I can access the site in a browser just fine. This is with a brand new, fresh install. This is what the log says.

```
[17:38:01.373] New session started.
[17:38:01.373] Software version: 5.0.1.
[17:38:01.373] Path: C:/Program Files (x86)/Grabber
[17:38:01.373] Loading preferences from C:/Users/Matt/AppData/Local/Bionus/Grabber/settings.ini
[17:38:01.375] Loading sources
[17:38:01.378] No site for source Gelbooru (0.1)
[17:38:01.382] %n source(s) found: behoimi.org (Danbooru), capi.sankakucomplex.com (Sankaku), chan.sankakucomplex.com (Sankaku), danbooru.donmai.us (Danbooru (2.0)), derpibooru.org (Booru-on-rails), e621.net (Danbooru), gelbooru.com (Gelbooru (0.2)), hijiribe.donmai.us (Danbooru (2.0)), ichijou.org (Danbooru), idol.sankakucomplex.com (Sankaku), konachan.com (Moebooru), rule34.paheal.net (Shimmie), rule34.xxx (Gelbooru (0.2)), safebooru.org (Gelbooru (0.2)), sonohara.donmai.us (Danbooru (2.0)), tentaclerape.net (Shimmie), www.zerochan.net (Zerochan), yande.re (Moebooru)
[17:38:01.383] Logging into chan.sankakucomplex.com (chan.sankakucomplex.com)...
[17:38:01.391] Updating settings for tab "Favorites".
[17:38:01.391] Updating checkboxes.
[17:38:01.407] End of initialization
[17:38:01.456] Loading error: Error creating SSL context ()
[17:38:01.456] Loading using Json failed. Retry using Html.
[17:38:01.524] Loading error: Error creating SSL context ()
[17:38:01.525] No valid source of the site returned result.
[17:38:01.525] Logging into chan.sankakucomplex.com (chan.sankakucomplex.com) finished (failure).
[17:38:01.528] Updating settings for tab "New tab".
[17:38:01.529] Updating checkboxes.
[17:38:08.888] Saving sources...
[17:38:08.888] Logging into chan.sankakucomplex.com (chan.sankakucomplex.com)...
[17:38:08.889] Updating checkboxes.
[17:38:08.891] Loading results...
[17:38:08.891] Loading page https://chan.sankakucomplex.com/post/index.json?limit=20&page=1&tags=
[17:38:08.952] Loading error: Error creating SSL context ()
[17:38:08.952] Loading using Json failed. Retry using Html.
[17:38:08.953] Loading error: Error creating SSL context ()
[17:38:08.953] Loading using Json failed. Retry using Html.
[17:38:08.958] Loading error: Error creating SSL context ()
[17:38:08.958] No valid source of the site returned result.
[17:38:08.959] Logging into chan.sankakucomplex.com (chan.sankakucomplex.com) finished (failure).
[17:38:08.959] Loading error: Error creating SSL context ()
[17:38:08.959] No valid source of the site returned result.
[17:38:13.639] Loading results...
[17:38:13.640] Loading page https://chan.sankakucomplex.com/post/index.json?limit=20&page=1&tags=1girl
[17:38:13.700] Loading error: Error creating SSL context ()
[17:38:13.701] Loading using Json failed. Retry using Html.
[17:38:16.581] Loading error: Error creating SSL context ()
[17:38:16.581] No valid source of the site returned result.
```

**Comment by Bionus** [comment]
From the `Error creating SSL context` log message, it seems like Grabber cannot connect using HTTPS.

Do you have these files in your installation folder? They should have been copied automatically by the installer:
- libeay32.dll
- libssl32.dll
- ssleay32.dll

**Comment by cybersoldier8** [comment]
Yes, all three files are in the installation folder.

**Comment by Bionus** [comment]
Do you have the `Microsoft Visual C++ 2015 Redistributable` installed? It should have been installed during the setup but maybe something weird happened.

If yes, I read that installing the `Microsoft Visual C++ 2008 SP1 Redistributable Package` also can help.

Strange, this runs fine on other setups. Which OS do you have installed? (I see a Windows x64) Is there anything special to it?

**Comment by cybersoldier8** [comment]
![image](https://cloud.githubusercontent.com/assets/24867372/21598153/85062886-d125-11e6-9c18-aec4d946c523.png)
I already have 2015 and 2008 (likely from installing various steam games.

This is on Windows 10 pro x64.

**Comment by NielR** [comment]
I' ve found a possible cause or effect of this: the url's are wrong. They display c.sankaku... instead of chan.sankaku...

![screen 1483644779](https://cloud.githubusercontent.com/assets/2423125/21694601/5c1bf3a2-d386-11e6-90be-75cf09ac9ddb.jpg)

**Comment by aarqon** [comment]
It's worth noting they also seem to try to load a .mp4 file every time as well (check the end of that URL given), which is usually not the case.
I also have this issue.

**Comment by aarqon** [comment]
Correction: I think you addressed that bit in #709 so it may not apply here.

**Comment by Bionus** [comment]
@NielR @aarqon your issues are related to #709, you can check the solution provided there to fix the problem:

> You can update your `model.xml` file (located in ` C:\Users\%USERNAME%\AppData\Local\Bionus\Grabber\sites\Sankaku`) with [this one](https://raw.githubusercontent.com/Bionus/imgbrd-grabber/0621bed01c04f2fa18f532d20617f8aa118d7de3/release/sites/Sankaku/model.xml), it should fix the issue until the next release.

As for OP's issue, I'm continuing investigations but it sure is strange.

**Comment by blitz9826** [comment]
Yo! Juts signed up to help a fellow (fapper) out. Here's what I did to get it to work:

First, I removed the three DLLs mentioned above.
Second, I installed this implementation of OpenSSL: http://slproweb.com/products/Win32OpenSSL.html (get 1.0.2k 32-bits)
Third, restart Grabber if it's open (or launch it)

Cheers!

**Comment by ravenlord1** [comment]
Mr. blitz9826, you are a gentleman and a scholar!  I was having the exact same issues as OP and spent hours bashing my brain and looking for a fix.  Yours was the one that worked perfectly and on the first try.  Tip of the hat to you, sir :)

**Comment by Bionus** [comment]
In that case I'll close this issue.
Note that OpenSSL 1.0.2k is shipped with Grabber since version 5.3.0.
